code/app/src/main/java/com/kernelcrew/moodapp/ui/FollowingFragment.java

```diff
@@ -0,0 +1,135 @@
+import android.widget.CheckBox;
+import android.widget.TextView;
+import androidx.annotation.NonNull;
+import androidx.fragment.app.Fragment;
+import androidx.navigation.Navigation;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.google.android.material.appbar.MaterialToolbar;
+import com.google.android.material.bottomnavigation.BottomNavigationView;
+import com.google.firebase.firestore.DocumentSnapshot;
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.kernelcrew.moodapp.R;
+import com.google.android.material.imageview.ShapeableImageView;
+import java.util.ArrayList;
+import java.util.List;
+
+public class FollowingFragment extends Fragment {
+
+    private RecyclerView followingRecyclerView;
+    private FollowingAdapter adapter;
+    private List<FollowingFragment.User> followingList = new ArrayList<>();
+    private FirebaseFirestore db;
+
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.fragment_following, container, false);
+
+        followingRecyclerView = view.findViewById(R.id.followingRecyclerView);
+        followingRecyclerView.setLayoutManager(new LinearLayoutManager(requireContext()));
+
+        adapter = new FollowingAdapter(followingList);
+        followingRecyclerView.setAdapter(adapter);
+
+        db = FirebaseFirestore.getInstance();
+        fetchFollowing();
+
+        BottomNavigationView bottomNavigationView = view.findViewById(R.id.bottom_navigation);
+        bottomNavigationView.setSelectedItemId(R.id.page_myProfile);
+        bottomNavigationView.setOnItemSelectedListener(item -> {
+            if (item.getItemId() == R.id.page_myProfile) {
+                Navigation.findNavController(view).navigate(R.id.myProfile);
+                return true;
+            }
+            return false;
+        });
+
+        MaterialToolbar topAppBar = view.findViewById(R.id.topAppBar);
+        topAppBar.setNavigationOnClickListener(v -> Navigation.findNavController(v).navigateUp());
+
+
+        return view;
+    }
+
+    private void fetchFollowing() {
+        db.collection("users").document("testUser").collection("following")
+                .get()
+                .addOnSuccessListener(queryDocumentSnapshots -> {
+                    followingList.clear();
+                    for (DocumentSnapshot doc : queryDocumentSnapshots) {
+                        String name = doc.getId();
+                        boolean isFollowed = doc.getBoolean("isFollowed") != null && doc.getBoolean("isFollowed");
+                        followingList.add(new User(name, isFollowed));
+                    }
+                    adapter.notifyDataSetChanged();
+                })
+                .addOnFailureListener(e -> System.out.println("TEST_FIRESTORE: Failed to load following."));
+    }
```

Comment:
This can also go in a controller.

The same comments re: filtering and "testUser" apply here too.